### PR TITLE
task: add actionlint to GH action

### PR DESCRIPTION
**Pull request by gssbzn**
inspired by https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1988

Add a linter for GH actions

Example run 
Screenshot 2024-03-05 at 11 55 00

### CODE REVIEW

File: .github/actionlint-matcher.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "problemMatcher": [
+    {
+      "owner": "actionlint",
+      "pattern": [
+        {
+          "regexp": "^(?:\\x1b\\[\\d+m)?(.+?)(?:\\x1b\\[\\d+m)*:(?:\\x1b\\[\\d+m)*(\\d+)(?:\\x1b\\[\\d+m)*:(?:\\x1b\\[\\d+m)*(\\d+)(?:\\x1b\\[\\d+m)*: (?:\\x1b\\[\\d+m)*(.+?)(?:\\x1b\\[\\d+m)* \\[(.+?)\\]$",
```
Comment (andreaangiolillo) [inline .github/actionlint-matcher.json:7]: [q] Why is this file needed? Could you add a comment in the file to provide a matching string?

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: .github/actionlint-matcher.json (hunk 1 above)
Comment (gssbzn) [reply .github/actionlint-matcher.json:7]: >Could you add a comment in the file to provide a matching string?

json doesn't support comments :) 

This files enables the GH annotations and used in the GH action like
```
echo "::add-matcher::.github/actionlint-matcher.json"
```
https://github.com/rhysd/actionlint/blob/main/docs/usage.md#problem-matchers

Comment (andreaangiolillo) [reply .github/actionlint-matcher.json:7]: oh nice thanks!
